paramiko/hostkeys.py

```diff
@@ -220,15 +227,15 @@ def __iter__(self):
     def __len__(self):
         return len(self.keys())
 
-    def __delitem__(self, key):
-        k = self[key]
-
     def __getitem__(self, key):
         ret = self.lookup(key)
         if ret is None:
             raise KeyError(key)
         return ret
 
+    def __delitem__(self, key):
+        pass  # Needed for instantiating HostKeys.
```

Comment:
Yea we couldn't figure out what the heck was going on with the original change (happened in the Python 3 shakeup). Originally we just nuked it but I guess Dorian found it was added on purpose. I don't really see why we couldn't make it actually do `del self[key]` tho...